lib/bigcommerce/index.ts

```diff
@@ -289,38 +295,46 @@ export async function addToCart(
     lines
   );
 
-  return bigcommerceToVercelCart(bigCommerceCart, productsByIdList, checkout, checkoutUrl);
+  return bigCommerceToVercelCart(bigCommerceCart, productsByIdList, checkout, checkoutUrl);
 }
 
 export async function removeFromCart(cartId: string, lineIds: string[]): Promise<VercelCart> {
   let cartState: { status: number; body: BigCommerceDeleteCartItemOperation };
-
-  for (let removals = lineIds.length; removals > 0; removals--) {
-    const lineId = lineIds[removals - 1]!;
-
-    const res = await bigcommerceFetch<BigCommerceDeleteCartItemOperation>({
+  const removeCartItem = async (itemId: string) => {
+    const res = await bigCommerceFetch<BigCommerceDeleteCartItemOperation>({
       query: deleteCartLineItemMutation,
       variables: {
         deleteCartLineItemInput: {
           cartEntityId: cartId,
-          lineItemEntityId: lineId
+          lineItemEntityId: itemId
         }
       },
       cache: 'no-store'
     });
 
-    cartState = res;
+    return res;
+  };
+
+   if (lineIds.length === 1) {
+    cartState = await removeCartItem(lineIds[0]!);
+   } else if (lineIds.length > 1) {
```

Comment:
Vercel API suggests that `lineIds` is `string[]`. For that reason we use `if/else` statement at the first place and now we have this check for more than 1 IDs.

 As I get it actually we can't [remove](https://github.com/bigcommerce/nextjs-commerce/blob/main/app/api/cart/route.ts#L23) more than one [line](https://github.com/bigcommerce/nextjs-commerce/blob/main/components/cart/delete-item-button.tsx#LL19C22-L19C22) at a time, so actually we always get array of one Id and I'm thinking if it's safe to simplify this logic by keeping only `cartState = await removeCartItem(lineIds[0]!)` or smth similar. 

As a safety measure we always can leave  it as it is but I don't see cases when `else if` block will be executed... 

@BC-krasnoshapka, what's your thought on this? 🤔

Reply:
I suggest we leave it as is now, because core expects it to be array, thus we have to support array.